src/main/java/net/fortuna/ical4j/model/component/Observance.java

```diff
@@ -155,12 +155,21 @@ public final Date getLatestOnset(final Date date) {
 
         if (initialOnset == null) {
             try {
-                initialOnset = applyOffsetFrom(calculateOnset(((DtStart) getProperty(Property.DTSTART)).getDate()));
+                DtStart dtStart = (DtStart) getProperty(Property.DTSTART);
```

Comment:
This could be replaces by Optional<DtStart>. Is there a reason you do not use Optional here?

I understand that getProperty do not return an Optional, and that refactoring the return statement of getProperty is not part of this bugfix. But I think one might benefit from such refactoring, since the method return is in fact optional.